Tests/TensorFlowTests/LayerTests.swift

```diff
@@ -360,6 +360,26 @@ final class LayerTests: XCTestCase {
         XCTAssertEqual(output, expected)
     }
 
+    func testAvgPool1DGradient() {
+        let layer = AvgPool1D<Float>(poolSize: 2, stride: 1, padding: .valid)
+        let x = Tensor(shape: [1, 4, 4], scalars: (0..<16).map(Float.init))
+        let computedGradient = gradient(at: x, layer) { $1($0).sum() }
+        // The expected value of the gradient was computed using the following Python code:
+        // ```
+        //   avgpool1D = tf.keras.layers.AvgPool1D(strides=1)
```

Comment:
You haven't mentioned the poolsize for the layer, I'm not sure if this will work. Same for the `avgpool2d` and `avgpool3d` example.

Reply:
@Shashi456  The default arguments of `pool_size` are `2`, `(2, 2)`, and `(2, 2, 2)` for the respective 1D, 2D, and 3D layers. At least this is the case with keras under tf 2.0.